lib/hashie/extensions/method_access.rb

```diff
@@ -35,6 +35,10 @@ def respond_to?(name, include_private = false)
       def method_missing(name, *args)
         return self[name.to_s] if key?(name.to_s)
         return self[name.to_sym] if key?(name.to_sym)
+        if name[-1] == '?'
```

Comment:
The above code seems to imply that `name` can be either a String or a Symbol, but this line only works for a String, so one or the other is probably wrong and needs a bit of research. 

If `name` can be both, this should probably look like this:

```ruby
name_s = name.to_s
return self[name_s] if key?(name.name_s)

name_sym = name.to_sym
return self[name_sym] if key?(name_sym)

return super unless name_s[-1] == '?'

name_without_question = name_s[0..-2]
key?(name_without_question) || key?(name_without_question.to_sym)
```